mona_sdk/client.py

```diff
@@ -250,8 +259,26 @@ def __init__(
             override_host=override_app_server_host
         )
         self.filter_none_fields_on_export = filter_none_fields_on_export
-        self._default_sampling_rate = default_sampling_rate
-        self._context_class_to_sampling_rate = context_class_to_sampling_rate or {}
+
+        # Data sampling.
+
+        self._sampling_config_name = sampling_config_name
+
+        sampling_config_list = (
+            self.get_sampling_factors() if sampling_config_name else []
+        )
+        sampling_config = sampling_config_list[0] if sampling_config_list else {}
+
+        # If sampling_config_name was provided, the client will be initiated with the
+        # sampling map saved to the index and the mapped default factor.
+
+        self._default_sampling_rate = (
+            sampling_config.get("default_factor") or default_sampling_rate
+        )
+
+        self._context_class_to_sampling_rate = (
+            sampling_config.get("factors_map") or context_class_to_sampling_rate
```

Comment:
old code had an "or {}" at the end which I think is required here too

Reply:
Why? it was moved to the top where we define SAMPLING_CONFIGURATION_DICT

Reply:
ha good catch - I missed it. The default for context_class_to_sampling_rate should be None, not empty dict - even though this is practically won't do any harm as long as you don't change the object, this is a common pitfall in python and is prone to errors. The default shouldn't be created in time of definition, but in time of execution so the object behavior will work as expected. For more info see here https://stackoverflow.com/questions/1132941/least-astonishment-and-the-mutable-default-argument
So just remove the default empty dict, so the default of context_class_to_sampling_rate will be None in the constructor; and add or {} either here or when you assign it into self._context_class_to_sampling_rate